tests/forms_tests/tests/test_forms.py

```diff
@@ -2377,3 +2377,46 @@ def test_baseform_repr_dont_trigger_validation(self):
         self.assertRaises(AttributeError, lambda: p.cleaned_data)
         self.assertFalse(p.is_valid())
         self.assertEqual(p.cleaned_data, {'first_name': 'John', 'last_name': 'Lennon'})
+
+    def test_field_name_with_hidden_input(self):
+        """
+        BaseForm._html_output() uses inconsistent formatting for normal row
+        """
+
+        class SomeForm(Form):
+            hidden1 = CharField(label='hidden1', widget=HiddenInput, initial='h1')
+
+            custom = CharField(label='custom')
+
+            hidden2 = CharField(label='hidden2', widget=HiddenInput, initial='h2')
+
+            def as_p(self):
+                return self._html_output(
+                    normal_row='<p%(html_class_attr)s>%(field)s %(field_name)s</p>',
+                    error_row='%s',
+                    row_ender='</p>',
+                    help_text_html=' <span class="helptext">%s</span>',
```

Comment:
Similarly, could this be an empty string to simply (please simplify other values if possible as well).